chain/net/toBed.go

```diff
@@ -0,0 +1,102 @@
+// are written to the slice when their last gap has been processed, the bed slice may not be position sorted. The RGB
+// color field corresponds to the level of Net.
+func ToBed(nets []Net) []bed.Bed {
+	var prevChrom string = nets[0].TName
+	var ans, currBed []bed.Bed
+	var currAnno []bedAnnotation
+	var highestLvl int
+	for i := range nets {
+		if nets[i].TName != prevChrom {
+			ans = append(ans, formatAll(currBed, currAnno)...)
+			currAnno = []bedAnnotation{}
+			currBed = []bed.Bed{}
+			highestLvl = 0
+		}
+		if nets[i].Class == "fill" {
+			if nets[i].Level > highestLvl {
+				highestLvl = nets[i].Level
+				currBed = append(currBed, bed.Bed{})
+				currAnno = append(currAnno, bedAnnotation{})
+			} else {
+				currAnno[nets[i].Level-1].blockSizes = append(currAnno[nets[i].Level-1].blockSizes, (currBed[nets[i].Level-1].ChromEnd-currBed[nets[i].Level-1].ChromStart)-currAnno[nets[i].Level-1].blockStarts[len(currAnno[nets[i].Level-1].blockStarts)-1])
+				currBed[nets[i].Level-1].Annotation = annoToStringSlice(currAnno[nets[i].Level-1], nets[i].Level)
+				ans = append(ans, currBed[nets[i].Level-1])
+			}
+			currBed[nets[i].Level-1] = createBed(nets[i])
+			currAnno[nets[i].Level-1] = bedAnnotation{thickStart: nets[i].TStart, thickEnd: nets[i].TStart, blockCount: 1, blockStarts: []int{0}, itemRGB: "0,0,0"}
+		} else {
+			currAnno[nets[i].Level-1].blockCount++
+			currAnno[nets[i].Level-1].blockSizes = append(currAnno[nets[i].Level-1].blockSizes, (nets[i].TStart-currBed[nets[i].Level-1].ChromStart)-currAnno[nets[i].Level-1].blockStarts[len(currAnno[nets[i].Level-1].blockStarts)-1])
+			currAnno[nets[i].Level-1].blockStarts = append(currAnno[nets[i].Level-1].blockStarts, (nets[i].TStart-currBed[nets[i].Level-1].ChromStart)+nets[i].TSize)
+		}
+		prevChrom = nets[i].TName
+	}
+	ans = append(ans, formatAll(currBed, currAnno)...)
+	return ans
+}
+
+func createBed(n Net) bed.Bed {
+	var b bed.Bed = bed.Bed{
+		Chrom:             n.TName,
+		ChromStart:        n.TStart,
+		ChromEnd:          n.TStart + n.TSize,
+		Name:              formatBedName(n),
+		Score:             n.Level,
+		Strand:            bed.Strand(parse.StrandToRune(n.Orientation)),
+		FieldsInitialized: 7,
+		Annotation:        []string{},
+	}
+	return b
+}
+
+func formatBedName(n Net) string {
+	return fmt.Sprintf("%s_%d_%d", n.QName, n.QStart, n.QStart+n.QSize)
+}
+
+func annoToStringSlice(a bedAnnotation, lvl int) []string {
+	switch lvl {
+	case 1:
+		a.itemRGB = "153,204,102"
+	case 2:
+		a.itemRGB = "246,237,100"
+	case 3:
+		a.itemRGB = "234,51,35"
+	case 4:
+		a.itemRGB = "124,199,216"
+	case 5:
+		a.itemRGB = "182,155,197"
+	default:
+		a.itemRGB = "0,0,0"
+	}
+	return []string{fileio.IntToString(a.thickStart), fileio.IntToString(a.thickEnd), a.itemRGB, fileio.IntToString(a.blockCount), fileio.IntSliceToString(a.blockSizes), fileio.IntSliceToString(a.blockStarts)}
+}
+
+func formatAll(beds []bed.Bed, anno []bedAnnotation) []bed.Bed {
```

Comment:
Can you add a quick comment here.  In code review we are having a tiny bit of trouble quickly figuring out the inputs and outputs.

Reply:
yes i can, for the formatAll function?

Reply:
Yes, please

Reply:
I added godocs to this function. Essentially, this function takes the bed struct that was just created, and appends the annotation field as a string from the orthologous annotation struct. The extra calculation is to get the last "block size" for the annotation field, which cannot be calculated until the program "realizes" that it has seen all lines of the file corresponding to a singular net struct